src/backend/distributed/commands/schema_based_sharding.c

```diff
@@ -117,6 +117,15 @@ ShouldCreateTenantSchemaTable(Oid relationId)
 		return false;
 	}
 
+	/*
+	 * Materialized views cannot be a tenant table but they are allowed to exist in
+	 * tenant schema.
+	 */
+	if (get_rel_relkind(relationId) == RELKIND_MATVIEW)
```

Comment:
what is specific about mat views?

Shouldn't it be:

```C
if (!RegularTable(relationId))
return false;
```

Reply:
I updated the place where check for mat views. Updated PR desription as well. The problem is that we allow mat views to be in distributed schema. But they should not be converted to a tenant table since they cannot be distributed.